diff --git a/schemas/sidebar.ts b/schemas/sidebar.ts
index 86eb7f87e48d2f1b2cb096efa7a5bfaae63939bf..d1cf4ca8f280082ebf1c41c8c849e838d2250f3d 100644
--- a/schemas/sidebar.ts
+++ b/schemas/sidebar.ts
@@ -49,6 +49,8 @@ const AutoSidebarGroupSchema = SidebarGroupSchema.extend({
 		// TODO: not supported by Docusaurus but would be good to have
 		/** How many directories deep to include from this directory in the sidebar. Default: `Infinity`. */
 		// depth: z.number().optional(),
+		sort: z.enum(['date']).optional(),
+		order: z.enum(['ascending','descending']).optional()
 	}),
 });
 export type AutoSidebarGroup = z.infer<typeof AutoSidebarGroupSchema>;
diff --git a/utils/navigation.ts b/utils/navigation.ts
index aa4ab1f4db2f4fcb6adb8122cadb89e3e05791b9..e4a85077b588de3e33cf64e7eb4fb91501fc1c13 100644
--- a/utils/navigation.ts
+++ b/utils/navigation.ts
@@ -91,7 +91,7 @@ function groupFromAutogenerateConfig(
 	routes: Route[],
 	currentPathname: string
 ): Group {
-	const { collapsed: subgroupCollapsed, directory } = item.autogenerate;
+	const { collapsed: subgroupCollapsed, directory, sort, order } = item.autogenerate;
 	const localeDir = locale ? locale + '/' + directory : directory;
 	const dirDocs = routes.filter(
 		(doc) =>
@@ -99,8 +99,9 @@ function groupFromAutogenerateConfig(
 			stripExtension(doc.id) === localeDir ||
 			// Match against `foo/anything/else.md`.
 			doc.id.startsWith(localeDir + '/')
-	);
-	const tree = treeify(dirDocs, localeDir);
+	)
+	const sorted = !sort ? dirDocs : dirDocs.sort(sortHandler(sort, order)).map((doc, i) => {doc.entry.data.sidebar.order = i; return doc});
+	const tree = treeify(sorted, localeDir);
 	return {
 		type: 'group',
 		label: pickLang(item.translations, localeToLang(locale)) || item.label,
@@ -110,6 +111,13 @@ function groupFromAutogenerateConfig(
 	};
 }
 
+const sortHandler = (kind: 'date', order: 'ascending' | 'descending') => {
+	if (kind === 'date') {
+		if (order === 'ascending') return (docA: Route, docB: Route) => docA.entry.data.date! > docB.entry.data.date! ? 1 : -1
+		return (docA: Route, docB: Route) => docA.entry.data.date! < docB.entry.data.date! ? 1 : -1
+	}
+}
+
 /** Check if a string starts with one of `http://` or `https://`. */
 const isAbsolute = (link: string) => /^https?:\/\//.test(link);
 
