diff --git a/dist/theme/url-builder.js b/dist/theme/url-builder.js
index dea04ef2259b55a0ccb9acc26b054cdc34d16054..466e9fd32b4354c2c314d8dfd081c3db61773f49 100644
--- a/dist/theme/url-builder.js
+++ b/dist/theme/url-builder.js
@@ -44,7 +44,7 @@ class UrlBuilder {
     getUrls(project) {
         var _a;
         const entryFileName = this.options.getValue('entryFileName');
-        const indexFileName = 'API.md';
+        const indexFileName = this.options.getValue('indexFileName');
         const hasReadme = !this.options.getValue('readme').endsWith('none');
         if (hasReadme) {
             this.urls.push(new typedoc_1.UrlMapping(entryFileName, project, this.theme.readmeTemplate));