diff --git a/dist/theme/resources/partials/table.properties.js b/dist/theme/resources/partials/table.properties.js
index 128ffeac571c2c0f1c3e4143965c201e90ccad8d..19477e03157bfc7a610d0d7718a5c3e95801a398 100644
--- a/dist/theme/resources/partials/table.properties.js
+++ b/dist/theme/resources/partials/table.properties.js
@@ -42,6 +42,7 @@ function propertiesTable(context, props, nameCol = 'Property') {
         if (property.flags.length && !property.flags.isOptional) {
             nameColumn.push(property.flags.map((flag) => (0, elements_1.backTicks)(flag.toLowerCase())).join(' '));
         }
+        nameColumn.push(`<a id="${property.anchor}" name="${property.anchor}"></a>`);
         if (Boolean(property.getSignature || Boolean(property.setSignature))) {
             nameColumn.push(context.declarationMemberAccessor(property));
         }
diff --git a/dist/theme/url-builder.js b/dist/theme/url-builder.js
index dea04ef2259b55a0ccb9acc26b054cdc34d16054..466e9fd32b4354c2c314d8dfd081c3db61773f49 100644
--- a/dist/theme/url-builder.js
+++ b/dist/theme/url-builder.js
@@ -44,7 +44,7 @@ class UrlBuilder {
     getUrls(project) {
         var _a;
         const entryFileName = this.options.getValue('entryFileName');
-        const indexFileName = 'API.md';
+        const indexFileName = this.options.getValue('indexFileName');
         const hasReadme = !this.options.getValue('readme').endsWith('none');
         if (hasReadme) {
             this.urls.push(new typedoc_1.UrlMapping(entryFileName, project, this.theme.readmeTemplate));